src/Workspaces/CSharp/Portable/Simplification/Reducers/CSharpVarReducer.Rewriter.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Utilities;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.Simplification
+{
+    internal partial class CSharpVarReducer
```

Comment:
this is now a nice and tiny type with the isolated logic to try to convert a TypeSyntax to 'var' during normal simplification (i.e. the Simplification subsystem, not the SimplifyTypeNames feature).  It's single-purpose and easy to understand, and isn't just interspersed randomly into other parts of the infrastructure.